python/pyspark/sql/dataframe.py

```diff
@@ -2305,16 +2305,15 @@ def _to_corrected_pandas_type(dt):
     uncorrectly.
     """
     import numpy as np
-    if type(dt) == ByteType:
-        return np.int8
-    elif type(dt) == ShortType:
-        return np.int16
-    elif type(dt) == IntegerType:
-        return np.int32
-    elif type(dt) == FloatType:
-        return np.float32
-    else:
-        return None
+    mappings = {
```

Comment:
I would also just keep the if-elif logic. It might be more efficient by using a map but it needs to create a map everytime this function is called. More importantly, this code path isn't supposed to be performance sensitive as it's called per a column. So, I would just keep the logic as was.